exercise-integration-testing-5-53/test/spread-the-code-for-task-10/when-event-is-valid.test.js

```diff
@@ -0,0 +1,270 @@
+  });
+});
+
+beforeEach(() => {
+    nock("http://localhost").get("/notification").reply(200, {
+      success: true,
+    });
+  });
+  
+  function helper(length) {
+    var randomChars =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    var result = "";
+    for (var i = 0; i < length; i++) {
+      result += randomChars.charAt(
+        Math.floor(Math.random() * randomChars.length)
+      );
+    }
+    return result;
+  }
+  describe("Sensors test", () => {
+    // ✅ TASK: Test that when a new event is posted to /event route, if category or temperature are not specified -> the API returns HTTP 400
+    // 💡 TIP: Down below, there is an example event schema
+    
+  
+    // ✅ TASK: Test that when a new event is posted to /sensor-events route, the temperature is not specified -> the event is NOT saved to the DB!
+    // 💡 TIP: Testing the response is not enough, the adequate state (e.g. DB) should also satisfy the expectation
+  
+    
+  
+    // ✅ TASK: Test that when a new valid event is posted to /sensor-events route, we get back a valid response
+    // 💡 TIP: Consider both the HTTP status and the body
+  
+    test("When new valid event is send,  should get back ", async () => {
+      //Arrange
+      const eventToAdd = {
+        category: "kids-room",
+        temperature: 20,
+        name: "Thermostat-temperature", //This must be unique
+        color: "Green",
+        weight: "80 gram",
+        status: "active",
+      };
+      //Act
+  
+      const config = {
+        validateStatus: () => true,
+      };
+      const url = `http://localhost:${port}/sensor-events`;
+      const res = await axios.post(url, eventToAdd, config);
+  
+      //Assert
+      expect(res.status).toBe(200);
+    });
+  
+    // ✅ TASK: Test that when a new valid event is posted to /sensor-events route, it's indeed retrievable from the DB
+    // 💡 TIP: Whenever possible, use the public API for verification
+  
+    test("When new valid event is send,  should retrievable the data from the DB (2 objects)", async () => {
+      //Arrange
+      const eventToAdd = {
+        category: "kids-room",
+        temperature: 20,
+        name: "Thermostat-temperature-new", //This must be unique
+        color: "Green",
+        weight: "80 gram",
+        status: "active",
+      };
+      //Act
+  
+      const config = {
+        validateStatus: () => true,
+      };
+      const urlFetchSensorEvent = `http://localhost:${port}/sensor-events`;
+      const urlGetEventByCategory = `http://localhost:${port}/sensor-events/${eventToAdd.category}/${eventToAdd.name}`;
+  
+      await axios.post(urlFetchSensorEvent, eventToAdd, config);
+      const eventFromDb = await axios.get(urlGetEventByCategory);
+      // console.log("retrive data ", eventFromDb);
+      //Assert
+      expect(eventFromDb.status).toBe(200);
+      expect(eventFromDb.data.length).toBe(2); // One obj from the last test and one is from the current test.
```

Comment:
⚠️ Important: The length is not the important thing here. With the same amount of lines it is possible more type of bugs - Whether the right events were retrieved

p.s. Why 2 items? You added only one